From 1cce053f18fd0b58358a859edc228de599246204 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Mon, 23 Jun 2025 20:10:14 +0800
Subject: [PATCH 09/63] loonggpu: Remove load callback from kms_driver

The ".load" callback in "struct drm_driver" is deprecated. In order to
remove the callback, we have to manually call "gsgpu_driver_load_kms"
instead.

Following up the reference implementation change.

Link: https://git.kernel.org/torvalds/c/90985660ba48
Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 loonggpu/loonggpu_drv.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/loonggpu/loonggpu_drv.c b/loonggpu/loonggpu_drv.c
index ae5d338..58d98c9 100644
--- a/loonggpu/loonggpu_drv.c
+++ b/loonggpu/loonggpu_drv.c
@@ -641,6 +641,8 @@ static int loongson_vga_pci_register(struct pci_dev *pdev,
 	if (pdev->device != 0x9a10 && loongson_gpu_pdev)
 		pci_set_drvdata(loongson_gpu_pdev, dev);
 
+	ret = loonggpu_driver_load_kms(dev, pci_gpu_flags);
+
 retry_init:
 	ret = drm_dev_register(dev, pci_gpu_flags);
 	if (ret == -EAGAIN && ++retry <= 3) {
@@ -739,7 +741,6 @@ static struct drm_driver kms_driver = {
 		| DRIVER_SYNCOBJ_TIMELINE
 #endif
 		| DRIVER_RENDER | DRIVER_ATOMIC,
-	.load = loonggpu_driver_load_kms,
 	.open = loonggpu_driver_open_kms,
 	.postclose = loonggpu_driver_postclose_kms,
 	lg_drm_driver_set_lastclose
-- 
2.52.0

